app/javascript/article-form/components/TagsField.jsx

```diff
@@ -1,34 +1,90 @@
 import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
 import PropTypes from 'prop-types';
-import { Tags } from '../../shared/components/tags';
+import { TagAutocompleteOption } from './TagAutocompleteOption';
+import { TagAutocompleteSelection } from './TagAutocompleteSelection';
+import { MultiSelectAutocomplete } from '@crayons';
+import { fetchSearch } from '@utilities/search';
 
-export const DEFAULT_TAG_FORMAT = '[0-9A-Za-z, ]+';
+/**
+ * TagsField for the article form. Allows users to search and select up to 4 tags.
+ *
+ * @param {Function} onInput Callback to sync selections to article form state
+ * @param {string} defaultValue Comma separated list of any currently selected tags
+ * @param {Function} switchHelpContext Callback to switch the help context when the field is focused
+ */
+export const TagsField = ({ onInput, defaultValue, switchHelpContext }) => {
+  const [defaultSelections, setDefaultSelections] = useState([]);
+  const [defaultsLoaded, setDefaultsLoaded] = useState(false);
+  const [topTags, setTopTags] = useState([]);
+
+  useEffect(() => {
+    window
+      .fetch('/tags/suggest')
+      .then((res) => res.json())
+      .then((results) => setTopTags(results));
+  }, []);
+
+  useEffect(() => {
+    // Previously selected tags are passed as a plain comma separated string
+    // Fetching furher tag data allows us to display a richer UI
+    // This fetch only happens once on first component load
+    if (defaultValue && defaultValue !== '' && !defaultsLoaded) {
+      const tagNames = defaultValue.split(', ');
+
+      const tagRequests = tagNames.map((tagName) =>
+        fetchSearch('tags', { name: tagName }).then(({ result }) => {
+          const potentialMatch = result[0] || {};
```

Comment:
nitpick (non-blocking): Prefer destructuring with a default value.

```suggestion
          const [ potentialMatch = {} ] = result;
```